Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,434 @@
+                }
+                if (gear.Children.Count > 0)
+                    AddEquipment(gear.Children);
+            }
+        }
+
+        private void RemoveEquipment(IList equipment)
+        {
+            foreach (Gear gear in equipment)
+            {
+                if (gear.Category == "Cyberdecks" || gear.Category == "Commlinks")
+                    Persons.Remove(gear);
+                else if (gear.Category.Contains("Program"))
+                    Programs.Remove(ProgramsList.Find(x => x.Name.Equals(gear.Name)));
+            }
+        }
+
+        private void Gear_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            if (e.NewItems != null)
+                AddEquipment(e.NewItems);
+            if (e.OldItems != null)
+                RemoveEquipment(e.OldItems);
+        }
+
+        private void character_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Character.WoundModifier))
+            {
+                OnPropertyChanged(nameof(WoundModifier));
+            }
+        }
+
+        public int GetTotalMatrixAttribute(string attribute)
+        {
+            switch (attribute)
+            {
+                case "Attack":
+                    return TotalAttack;
+                case "Sleaze":
+                    return TotalSleaze;
+                case "Data Processing":
+                    return TotalDataProcessing;
+                case "Firewall":
+                    return TotalFirewall;
+            }
+            return 0;
+        }
+
+        public int GetTotalSkill(string skillName)
+        {
+            Skill skill = _character.SkillsSection.GetActiveSkill(skillName);
+            if (skill != null)
+                return skill.TotalBaseRating;
+            return 0;
+        }
+
+        public int GetTotalAttribute(string attributeName)
+        {
+            CharacterAttrib attribute = _character.GetAttribute(attributeName);
+            if (attribute != null)
+                return attribute.TotalValue;
+            return 0;
+        }
+
+        /// <summary>
+        /// Parse string as integer or as character attribute by attribute's name
+        /// </summary>
+        /// <param name="value">Integer value or Attribute's name</param>
+        /// <returns></returns>
+        private int Parse(string value)
+        {
+            int result;
+            if (!int.TryParse(value, out result))
+            {
+                CharacterAttrib attribute = _character.GetAttribute(value.Replace("{", "").Replace("}", ""));
+                if (attribute != null)
+                    result = attribute.Base;
+            }
+            return result;
+        }
+
+        #endregion
+
+        #region Properties
+
+        public BindingList<Gear> Persons { get; set; }
+        public BindingList<Program> Programs { get; set; }
```

Comment:
Do these need to be `BindingList<T>`s? `BindingList<T>` is a fairly old and troublesome class that really should only be used in certain situations where some features specific to it are vital for how some code works (mainly stuff related to the specific order of items in the list). If you just want collections that have events registered to them when contents are changed, you should use `ObservableCollection<T>` instead. And if you aren't registering any events like these at all, you should just use a normal collection like `List<T>` or `HashSet<T>` instead.